[RHACS] ROX-10853: Update documentation on reissuing certificates (#46387)
openshift/openshift-docs: modules/reissue-internal-certificates-secured-clusters.adoc

// Module included in the following assemblies:
//
// * configuration/reissue-internal-certificates.adoc
:_module-type: PROCEDURE
[id="reissue-internal-certificates-secured-cluster_{context}"]
= Reissuing internal certificates for Secured Clusters using init bundles

Secured clusters contain the Collector, Sensor, and Admission Control components. These components use a built-in server certificate for authentication when communicating with other {product-title} components.

The RHACS portal shows an information banner when the Central certificate is about to expire.

[NOTE]
====
The information banner only appears 15 days before the certificate expiry date.
====

.Prerequisites

* To reissue certificates, you must have `write` permission for the `ServiceIdentity` resource.

.Procedure

. Choose *Platform Configuration* -> *Clusters*.
. Click *Manage Tokens*.
. Navigate to the *Authentication Tokens* section and click on *Cluster Init Bundle*.
. Click *Generate bundle*.
. Enter a name for the cluster init bundle and click *Generate*.

> . Click *Download Kubernetes secrets file* to download the generated bundle.

Author: Thank you, I will add this!

// Module included in the following assemblies:
//
// * configuration/reissue-internal-certificates.adoc
:_module-type: PROCEDURE
[id="reissue-internal-certificates-secured-cluster_{context}"]
= Reissuing internal certificates for Secured Clusters using init bundles

Secured clusters contain the Collector, Sensor, and Admission Control components. These components use a built-in server certificate for authentication when communicating with other {product-title} components.

The RHACS portal shows an information banner when the Central certificate is about to expire.

[NOTE]
====
The information banner only appears 15 days before the certificate expiry date.
====

.Prerequisites

* To reissue certificates, you must have `write` permission for the `ServiceIdentity` resource.

[IMPORTANT]
====
Store this bundle securely because it contains secrets.
You can use the same bundle on multiple secured clusters.
====

.Procedure

> To generate an init bundle using the {product-title-short} portal:

Reviewer: ```suggestion
To generate an `init` bundle using the {product-title-short} portal:
```
Use backticks for all non-GUI "system items", including daemon, service, or software package. [Quick markup reference](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference)

Author: Hi, throughout the docs they have previously been called init bundle (no backticks). The product was originally an acquisition so that may be why it doesn't use RH/IBM style. I am reluctant to change this for this release note, since it would be inconsistent with how it is referred to elsewhere in the docs.

> . Choose *Platform Configuration* -> *Clusters*.

Reviewer: ```suggestion
. Select *Platform Configuration* -> *Clusters*.
```
Per ISG:
https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#verbs-to-use

> . Click *Manage Tokens*.

Reviewer: Global comment for the use of the term "Click"
Per ISG: https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#verbs-to-use

> If you write content that might be viewed both on mobile and non-mobile devices and using _click_ might cause confusion, you can use generic verbs such as select.

Author: Due to the use of "click" elsewhere in doc, and because it's desktop-only, I would prefer to use click to be consistent until we have a chance to further clean up the docs.

> . Navigate to the *Authentication Tokens* section and click on *Cluster Init Bundle*.

Reviewer: ```suggestion
. Navigate to the *Authentication Tokens* section, and click the *Cluster Init Bundle*.
```
Per ISG: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-commas#commas-between-clauses

> Use a comma between independent clauses that are separated by a coordinating conjunction unless the clauses are short or closely related. Coordinating conjunctions are and, but, or, nor, for, so, and yet.

. Click *Generate bundle*.

> . Enter a name for the cluster init bundle and click *Generate*.

Reviewer: ```suggestion
. Enter a name for the cluster `init` bundle, and click *Generate*.
```

. Click *Download Kubernetes secrets file* to download the generated bundle.

> To generate an init bundle using the `roxctl` CLI, enter the following command:

Reviewer: ```suggestion
To generate an `init` bundle using the `roxctl` CLI, enter the following command:
```

[source,terminal]
----
$ roxctl -e <endpoint> -p <admin_password> central init-bundle generate <bundle_name> --output-secrets init-bundle.yaml
----

.Next steps

> * Use the `oc -n stackrox apply -f <init-bundle.yaml>` command to create the necessary resources on each secured cluster.

Reviewer: ```suggestion
* To create the necessary resources on each secured cluster, run the following command:
[source,terminal]
oc -n stackrox apply -f <init-bundle.yaml>
```
Per the OpenShift Docs guidance on [Code blocks, command syntax, and example output](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output):

> - If a step in a procedure is to run a command, make sure that the step text includes an explicit instruction to "run" or "enter" the command. In most cases, use one of the following patterns to introduce the code block:
>  _Step description_, run the following command:
>  _Step description_, enter the following command:
> 
> - Do NOT use any markup in code blocks; code blocks generally do not accept any markup.

// Module included in the following assemblies:
//
// * configuration/reissue-internal-certificates.adoc
:_module-type: PROCEDURE
[id="reissue-internal-certificates-secured-cluster_{context}"]
= Reissuing internal certificates for Secured Clusters using init bundles

Secured clusters contain the Collector, Sensor, and Admission Control components. These components use a built-in server certificate for authentication when communicating with other {product-title} components.

> The RHACS portal shows an information banner when the Central certificate is about to expire.

Reviewer 2: ```suggestion
The {product-title-short} portal shows an information banner when the Central certificate is about to expire.
```
(or whatever is right)

[NOTE]
====
The information banner only appears 15 days before the certificate expiry date.
====

.Prerequisites

* To reissue certificates, you must have `write` permission for the `ServiceIdentity` resource.

[IMPORTANT]
====
Store this bundle securely because it contains secrets.
You can use the same bundle on multiple secured clusters.
====

.Procedure

> To generate an init bundle using the {product-title-short} portal:

Reviewer 2: Since you have two ways to do this (iiuc), probably stick a bullet on this line and on line 38 (CLI method). Then make the `.` steps into `..` steps under the `*`

. Select *Platform Configuration* -> *Clusters*.
. Click *Manage Tokens*.
. Navigate to the *Authentication Tokens* section, and click *Cluster Init Bundle*.
. Click *Generate bundle*.
. Enter a name for the cluster init bundle and click *Generate*.

> . Click *Download Kubernetes secrets file* to download the generated bundle.

Reviewer 2: Just a suggestion, I think it works either way but I'm biased towards goal-first language :)

```suggestion
. To download the generated bundle, click *Download Kubernetes secrets file*.
```

To generate an init bundle using the `roxctl` CLI, run the following command:

[source,terminal]
----
$ roxctl -e <endpoint> -p <admin_password> central init-bundle generate <bundle_name> --output-secrets init-bundle.yaml
----

.Next steps

* To create the necessary resources on each secured cluster, run the following command:

[source,terminal]
----
$ oc -n stackrox apply -f <init-bundle.yaml>
----

[role="_additional-resources"]
.Additional resources

> * xref:../installing/install-ocp-operator.adoc#create-resource-init-bundle[Creating resources by using the init bundle]

Reviewer 2: These lines need to be removed from the module and added to the assembly right under the `include` for this module.
